### PR TITLE
feat: add flows argument

### PR DESCRIPTION
**Pull request by Julieta11**

### CODE REVIEW

File: rust/src/client.rs [hunk 1]
```diff
@@ -1,23 +1,248 @@
...
+        &self,
+        module: &FriendshipsServiceClient<WebSocketTransport>,
+        token_user_a: &str,
+        token_user_b: &str,
+        user_a_address: &str,
+        user_b_address: &str,
+    ) {
+        match self {
+            Flow::Flow1 => {
+                // Implement Flow 1: Request A-B -> Cancel A-B
```
Comment (Julieta11) [inline rust/src/client.rs:43]:
```suggestion
                // Implement Flow 1: Request A-B, Cancel A-B
```

File: rust/README.md [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
-#### Client A: Executor
-This is the executor of the actions, gets all friends and requests and then sends a friendship request and a cancel.
-`cargo run --bin a-client-ws`
+### Client A: Executor
+This is the executor of the actions, gets all friends and requests and then performs actions based on the specified flow. 
+
+There are two available flows:
+
+- `flow1`: Sends a friendship request and then cancels it.
+- `flow2`: Sends a friendship request, accepts the friendship, and then deletes it.
```
Comment (Julieta11) [inline rust/README.md:12]:
```suggestion
- `flow2`: User A sends a friendship request, User B accepts the friendship, and then User A deletes it.
```

File: rust/README.md [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
-This is the executor of the actions, gets all friends and requests and then sends a friendship request and a cancel.
-`cargo run --bin a-client-ws`
+### Client A: Executor
+This is the executor of the actions, gets all friends and requests and then performs actions based on the specified flow. 
+
+There are two available flows:
+
+- `flow1`: User A sends a friendship request and then cancels it.
+- `flow2`: User A sends a friendship request, User B accepts the friendship, and then User A deletes it.
+- `flow3`: Sends a friendship request and then rejects it.
```
Comment (Julieta11) [inline rust/README.md:13]:
```suggestion
- `flow3`: User A sends a friendship request and then User B rejects it.
```

File: rust/src/client.rs [hunk 4]
```diff
@@ -1,23 +1,289 @@
...
+        Err(err) => {
+            panic!("{err:?}")
+        }
+    }
+}
+
 #[tokio::main]
 async fn main() {
-    let token = "";
+    let token_user_a = "";
```
Comment (agusaldasoro) [inline rust/src/client.rs:277]: can we remove these hardcoded values and read them from std-in instead?

File: rust/src/client.rs [hunk 5]
```diff
@@ -1,23 +1,289 @@
...
+            Flow::Flow3 => {
+                // Implement Flow 3: Request A-B, Reject B-A
+                request_a_to_b(module, token_user_a, user_b_address).await;
+                reject_b_to_a(module, token_user_b, user_a_address).await;
+            }
+            Flow::Flow4 => {
+                // Implement Flow 4: Request A-B, Accept B-A, Delete B-A
+                request_a_to_b(module, token_user_a, user_b_address).await;
+                accept_b_to_a(module, token_user_b, user_a_address).await;
+                delete_b_to_a(module, token_user_a, user_b_address).await;
```
[the remark this answers is not in the document]
Comment (Julieta11) [reply rust/src/client.rs:64]: hahaha right 🤓

**Review by agusaldasoro — APPROVED** [review]
Beautiful!

**Pull request merged**